rts/Lua/LuaConstGame.cpp

```diff
@@ -28,6 +28,13 @@
  * @see rts/Lua/LuaConstGame.cpp
 ******************************************************************************/
 
+/***
+ * @class TextColorCodes
```

Comment:
@rhys-vdw a type like this, reused in different files needs to be redeclared? If not what can we establish as convention for?

Reply:
- It does not and should not be redeclared. If it is declared twice then the fields will be doubled up on the definition.
- If there is one obvious API that uses the type, then declare it next to it (so it's easy to update them together.
- If it's used by many functions shared between different files (e.g. `CommandDescription`) or has no obvious home, then my choice was to declare it in a `library/*.lua` file.

In this case, I wonder if that duplicate code can be moved into a single function `PushColorCodes` containing [this logic](https://github.com/beyond-all-reason/spring/pull/1988/files/06bd2fe33a0dd6805a7b024104542d895979f580#diff-85b82291d073a0f5bf8b0a4d29693156937cb4359544e2816acc61f3980a21e9R335-R341) and then the class definition can be declared next to that.

Reply:
@technateNG Do you mind removing this and the corresponding declaration in `Engine.cpp` and adding a single one in `library/Types.lua`?

Would look something like this:

```lua
---Contains data about font rendering colors.
---
---@class TextColorCode
---@field Color string Indicates that the following bytes contain color code information
---@field ColorAndOutline string Indicates that the following bytes contain color code and outline information
---@field Reset string Indicates reset of the current color
```

Thank you!

Reply:
> Eventually the copy in the `Game.` table would ideally be removed so perhaps just keep the doc entry near the `Engine.` one and keep the one in `Game.` silently referencing the type.

> @badosu: "...removing this and the corresponding declaration in Engine.cpp..."

Yes, both need to be removed and added as a single reference in `Types.lua`.

Reply:
> Yes, both need to be removed and added as a single reference in Types.lua.

What I meant is that at some point the actual Lua table entries (the `luaL_createtable` and `lua_pushnumber`s) would be removed.

Reply:
I understand now, makes sense.

Reply:
@badosu Done.